src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,46 @@
+.calendar {
+  display: flex;
+  gap: $gap;
+  flex-wrap: wrap;
+  padding: 10px;
+  width: $cols-number * $cell-size + $gap * ($cols-number - 1);
+
+  @for $i from 28 through 31 {
+    &--month-length--#{$i} :nth-child(#{$i}) ~ * {
+      display: none;
+    }
+  }
+
+  &--start-day {
+    @each $week-day, $shift in $week-days {
+      &--#{$week-day} :first-child {
+        margin-left: ($cell-size + $gap) * $shift;
+      }
+    }
+  }
+
+  &__day {
+    box-sizing: border-box;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: $cell-size;
+    width: $cell-size;
+    background-color: #eee;
```

Comment:
It is a good practice to make all colors [variable](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties)